### ISSUE TITLE
authplain: configuration shows unsupported hashing algorithms / password reset status 500

### ISSUE THREAD

**Issue by mchubby** [splitbrain/dokuwiki#1748]
Hello,
The authplain algo list ``$meta['passcrypt']`` is populated with a static list of password derivation algorithms/hashing methods.

[Edit: 2016-06-26a "Elenor of Tsort"]

It should be filtered to only show algorithms supported by PassHash.class.php in the configuration dropdown control, and/or a validation should occur when the form is submitted.

**Symptoms:**

Password reset runs fairly normally, until mail reset link lands on blank page, with error.log

> 
```
PHP Fatal error:  Uncaught exception 'Exception' with message 'This PHP installation has no PBKDF2 support' in <redacted>/inc/PassHash.class.php:471
Stack trace:
#0 <redacted>/inc/PassHash.class.php(492): PassHash->hash_djangopbkdf2('<redacted>~', NULL, Array)
#1 <redacted>/inc/auth.php(1266): PassHash->hash_djangopbkdf2_sha256('<redacted>~', NULL)
#2 <redacted>/lib/plugins/authplain/auth.php(192): auth_cryptPassword('<redacted>~')
#3 [internal function]: auth_plugin_authplain->modifyUser('<redacted>', Array)
#4 <redacted>/lib/plugins/auth.php(135): call_user_func_array(Array, Array)
#5 <redacted>/inc/auth.php(1180): DokuWiki_Auth_Plugin->triggerUserMod('modify', Array)
#6 <redacted>/inc/actions.php(94): act_resendpwd()
#7 <redacted>/doku.php(124): act_dispatch()
#8 {main}
  thrown in <redacted>/inc/PassHash.class.php on line 471
```

Setting a new password via the profile page or via user management UI also yield a similar 500 error.

**Comment by splitbrain** [comment]
What PHP version are you using?

**Comment by mchubby** [comment]
It is running the latest wheezy package [php5 (5.4.45-0+deb7u5)](https://packages.debian.org/fr/wheezy/php5)

**Comment by phy25** [comment]
According to https://www.php.net/manual/en/function.hash-pbkdf2.php it's only available on PHP 5.5+, while the author had PHP 5.4. The latest DokuWiki version now requires PHP 5.6+ so I will close out the issue.
